source/main.cpp

```diff
@@ -355,30 +361,36 @@ std::string getInput(HB_Keyboard* sHBKB, bool &bCancelled)
         if (KBState == 3)
         {
             bCancelled = true;
+            input = "";
             break;
         }
         // Otherwise if the user has entered a key
         else if (KBState != 4)
         {
-            printf("%c[2K\r", 27);
-            printf("%s", input.c_str());
+            if(strcmp(last_input.c_str(),input.c_str()) != 0) {
+                ui_menu_draw_string(input.c_str(), 10, 10, COLOR_SELECTED);
+                sceneDraw();
+                last_input = input;
+            }
+            //Wait for VBlank
```

Comment:
Yes; it's one of my pet hates right now. I need to look into the routines around it a bit more. I assumed that maybe I hadn't switched to the correct video mode or something similar.